reference-architecture/aws-ansible/playbooks/roles/cloudformation-infra/tasks/main.yaml

```diff
@@ -0,0 +1,135 @@
+---
+- name: Create Greenfield Infrastructure
+  cloudformation:
+    stack_name: "OpenShift-Infra"
+    state: "present"
+    region: "{{ region }}"
+    template: "roles/cloudformation-infra/files/greenfield.json"
+    template_parameters:
+      Region: "{{ region }}"
+      Route53HostedZone: "{{ public_hosted_zone }}."
+      PublicHostedZone: "{{ public_hosted_zone }}"
+      MasterApiPort: "{{ console_port }}"
+      MasterHealthTarget: "TCP:{{ console_port }}"
+      MasterClusterHostname: "{{ openshift_master_cluster_hostname }}"
+      MasterClusterPublicHostname: "{{ openshift_master_cluster_public_hostname }}"
+      AppWildcardDomain: "*.{{ wildcard_zone }}"
+      VpcCidrBlock: "{{ cidr_block }}"
+      VpcName: "{{ vpc_prefix }}"
+      SubnetAvailabilityZones: "{{ vpc_subnet_azs }}"
+      SubnetCidrBlocks: 10.20.1.0/24,10.20.2.0/24,10.20.3.0/24,10.20.4.0/24,10.20.5.0/24,10.20.6.0/24
+      KeyName: "{{ keypair }}"
+      MasterInstanceType: "{{ master_instance_type }}"
+      AmiId: "{{ ami }}"
+      BastionInstanceType: "{{ node_instance_type }}"
+      MasterRootVolSize: 10
+      BastionRootVolType: gp2
+      MasterRootVolType: gp2
+      MasterDockerVolSize: "{{ docker_storage }}"
+      MasterDockerVolType: gp2
+      MasterEtcdVolSize: "{{ etcd_storage }}"
+      MasterEtcdVolType: gp2
+      InfraInstanceType: "{{ node_instance_type }}"
+      InfraRootVolSize: 25
+      InfraRootVolType: gp2
+      InfraDockerVolSize: "{{ docker_storage }}"
+      InfraDockerVolType: gp2
+      NodeEmptyVolSize: "{{ emptydir_storage }}"
+      NodeEmptyVolType: gp2
+      AppNodeInstanceType: "{{ node_instance_type }}"
+      NodeRootVolSize: 25
+      NodeRootVolType: gp2
+      NodeDockerVolSize: "{{ docker_storage }}"
+      NodeDockerVolType: gp2
+  when: create_vpc == "yes"
+
+- name: Create Brownfield Infrastructure
+  cloudformation:
+    stack_name: "OpenShift-Infra-Brownfield"
+    state: "present"
+    region: "{{ region }}"
+    template: "roles/cloudformation-infra/files/brownfield.json"
+    template_parameters:
+      Vpc: "{{ vpc_id }}"
+      PrivateSubnet1: "{{ private_subnet_id1 }}"
+      PrivateSubnet2: "{{ private_subnet_id2 }}"
+      PrivateSubnet3: "{{ private_subnet_id3 }}"
```

Comment:
In the reference arch document I called out that a region must have at least 3 or more AZs.  I haven't put forth the time yet to figure out a long term plan for zones with only 2 AZs. Most likely it would be spawn all masters in 1 AZ because if an AZ would go down we would have split brain. I Just don't think we are ready to facilitate this yet.  

In the future, we would like to move to ASGs but this work mainly just migrated the ansible deployed infrastructure as is to support the refarch document as is.  For the 3.3/3.4 release we will be diving more into that work.